src/lib/axios.ts

```diff
@@ -49,13 +59,35 @@ export class AxiosManager {
   }
 
   private async rejectResponseInterceptor(
-    error: AxiosError<{ errors: string[] }>
+    error: AxiosError<{ message: string }>
   ) {
     const originalRequest = error.config
+    const errorMessage = error?.response?.data.message
+
     if (!originalRequest) {
       return Promise.reject(error)
     }
-    //TODO: handle error when accessToken invalid
+
+    if (
+      errorMessage?.includes('token is expired') ||
+      errorMessage?.includes('token is invalid')
+    ) {
+      try {
+        const { data } = await this.axiosRefresh.post<
+          any,
+          { data: CreateAuthRefreshTokenReesponse }
+        >(`v2/corporate/auth/refresh`)
+
+        store.dispatch(validAuth(data.data))
+        localStorage.setItem('accessToken', data.data.token)
+      } catch (error) {
+        localStorage.removeItem('accessToken')
+        store.dispatch(invalidAuth('User session expired, please login again'))
+        persistor.flush().then(() => {
+          return persistor.purge()
+        })
```

Comment:
persistor.flush tuh buat apa dah? kalo itu ngapus localStorage berarti dispatch invalid nya useless mending dihapus aja

```suggestion
      } catch (error) {
        localStorage.removeItem('accessToken')
        persistor.flush().then(() => persistor.purge())
        return Promise.reject(error)
```

Reply:
ini buat ngehapus redux yang di persist ke initial statenya

Reply:
biar redux persist state yang di localStorage ke reset, ngehapus data user yang di persist